rpc/core/env.go

```diff
@@ -165,16 +165,15 @@ func InitGenesisChunks() error {
 	if err != nil {
 		return err
 	}
-
+	mut.Lock()
```

Comment:
[optional] It would be safer to introduce a helper function responsible for updating or amending `globalEnv` which also acquires/releases the lock.
[question] Also, I'm curious about why we didn't release the lock immediately after the append operation. Is there a specific reason for locking it prior to the loop and not locking it for each individual amendment?

Reply:
good point with the getter and setters, that's what we should do I think if we do want to try to make the global env more thread safe. the current locks we're providing are not very meaningful afaiu
https://github.com/celestiaorg/celestia-core/blob/22012af697b3e0578bef60f8c249d03b60f01606/rpc/core/env.go#L52-L56

but I'm unsure of actually fixing it short of a redesign of the rpc to have multiple instances work in process. 

> I'm curious about why we didn't release the lock immediately after the append operation.

this is also a good option, however since the loop is just iterating over some data and adding it to the chunks, this seemed like a good point to stop any other access to it while we write.

Reply:
> this is also a good option, however since the loop is just iterating over some data and adding it to the chunks, this seemed like a good point to stop any other access to it while we write.

Make sense! In that case, alternatively, we could define a dedicated function that receives a set of data and appends it to the global environment while also locking the mut (mutex), it would be cleaner and less error-prone (totally optional though).